### PR TITLE
feat(judicial-system): S3 Iteration

### PR DESCRIPTION
**Pull request by gudjong**
# S3 Iteration

https://app.asana.com/0/1199153462262248/1201777996978574/f

## What

- No longer removes case files from S3 when they have been uploaded to court.
- A case file can be viewed in the client until the case has been completed and the case file uploaded to court.

## Why

Usability improvements.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/judicial-system/backend/src/app/modules/file/file.service.ts [hunk 1]
```diff
@@ -220,22 +215,22 @@ export class FileService {
       courtCaseNumber,
     )
 
-    const documentId = await this.throttle
+    await this.throttle
 
-    const s3Key = file.key
-
-    const [nrOfRowsUpdated] = await this.fileModel.update(
-      { state: CaseFileState.STORED_IN_COURT, key: documentId },
```
Comment (Arnarkari93) [inline apps/judicial-system/backend/src/app/modules/file/file.service.ts:228]: So are we saying that we keep the same `case_file.key` after uploading to court? No need to get the `documentId` anymore?

File: apps/judicial-system/web/src/components/CaseFileList/CaseFileList.tsx [hunk 2]
```diff
@@ -44,17 +46,19 @@ const CaseFileList: React.FC<Props> = (props) => {
               file={file as TCaseFile}
               showFileSize={true}
               defaultBackgroundColor={
-                file.state === CaseFileState.BOKEN_LINK ? 'dark100' : 'blue100'
+                file.status === 'broken' ? 'dark100' : 'blue100'
```
Comment (Arnarkari93) [inline apps/judicial-system/web/src/components/CaseFileList/CaseFileList.tsx:49]: I was a bit confused reading this. First I read `file.status` as `file.state` and I did not know where this 'broken' was defined. Then I figured out these were two different properties of the file.

I want to suggest that we rename the `file.state` to `file.storage` with the options `IN_COURT | IN_RVG | DELETED`
we can keep the `file.status` as it is.

Just a suggestion 🤓

File: apps/judicial-system/backend/src/app/modules/file/file.service.ts (hunk 1 above)
Comment (gudjong) [reply apps/judicial-system/backend/src/app/modules/file/file.service.ts:228]: Yes, we have never used the documentId we get from Auður, so I decided to just drop it.

**Review by Arnarkari93 — APPROVED** [review]
LGTM

File: apps/judicial-system/web/src/components/CaseFileList/CaseFileList.tsx (hunk 2 above)
Comment (gudjong) [reply apps/judicial-system/web/src/components/CaseFileList/CaseFileList.tsx:49]: I agree, this is somewhat confusing. However, renaming the `state` field is a rather big change, it comes all the way from the database. On the other hand `status` is defined in the client and could be easily changed to something different, for instance `uploadStatus`. I will create a task in the Asana inbox for refactoring this.

**Pull request merged**